src/applications/Utilities/src/Utilities.cxx

```diff
@@ -644,59 +645,33 @@ int algorithmsRunner()
     using DefaultImageType = itk::Image< unsigned int, TImageType::ImageDimension >;
     auto inputImage = cbica::ReadImage< DefaultImageType >(inputImageFile);
     auto referenceImage = cbica::ReadImage< DefaultImageType >(referenceMaskForSimilarity);
-    auto uniqueLabels = cbica::GetUniqueValuesInImage< DefaultImageType >(inputImage);
-    auto uniqueLabelsRef = cbica::GetUniqueValuesInImage< DefaultImageType >(referenceImage);
 
-    // sanity check
-    if (uniqueLabels.size() != uniqueLabelsRef.size())
-    {
-      std::cerr << "The number of unique labels in input and reference image are not consistent.\n";
-      return EXIT_FAILURE;
-    }
-    else
+    auto stats = cbica::GetLabelStatistics< DefaultImageType >(inputImage, referenceImage);
+
+    std::cout << "Metric,Value\n";
+    for (const auto &stat : stats)
     {
-      for (size_t i = 0; i < uniqueLabels.size(); i++)
-      {
-        if (uniqueLabels[i] != uniqueLabelsRef[i])
-        {
-          std::cerr << "The label values in input and reference image are not consistent.\n";
-          return EXIT_FAILURE;
-        }
-      }
+      std::cout << stat.first << "," << stat.second << "\n";
     }
 
-    auto similarityFilter = itk::LabelOverlapMeasuresImageFilter< DefaultImageType >::New();
+    return EXIT_SUCCESS;
+  }
+  else if (requestedAlgorithm == LabelSimilarityBraTS)
+  {
+    // this filter only works on unsigned int type
+    using DefaultImageType = itk::Image< unsigned int, TImageType::ImageDimension >;
+    auto inputImage = cbica::ReadImage< DefaultImageType >(inputImageFile);
+    auto referenceImage = cbica::ReadImage< DefaultImageType >(referenceMaskForSimilarity);
 
-    similarityFilter->SetSourceImage(inputImage);
-    similarityFilter->SetTargetImage(referenceImage);
-    similarityFilter->Update();
+    auto stats = cbica::GetBraTSLabelStatistics< DefaultImageType >(inputImage, referenceImage);
 
-    //std::cout << "=== Entire Masked Area ===\n";
-    std::cout << "Property,Value\n";
-    std::cout << "TotalOverlap," << similarityFilter->GetTotalOverlap() << "\n";
-    std::cout << "Union(Jaccard)_Overall," << similarityFilter->GetUnionOverlap() << "\n";
-    std::cout << "Mean(DICE)_Overall," << similarityFilter->GetMeanOverlap() << "\n";
-    std::cout << "VolumeSimilarity_Overall," << similarityFilter->GetVolumeSimilarity() << "\n";
-    std::cout << "FalseNegativeError_Overall," << similarityFilter->GetFalseNegativeError() << "\n";
-    std::cout << "FalsePositiveError_Overall," << similarityFilter->GetFalsePositiveError() << "\n";
-
-    if (uniqueLabels.size() > 2) // basically if there is something more than 0 and 1
```

Comment:
Maybe you can also consider the case where the label image has labels i.e. 1 & 2 which cover the whole range of the image. In that case you might want stats for individual labels, because the total overlap above would be 100% regardless.

Reply:
I don't understand what you are saying, here. The stats are given for all labels individually (excluding background), regardless.

Reply:
You only show individual stats if there are > 2 labels and you presume that these two labels would be 0 and X. I presented an edge case where this labels would be X and Y, where X != 0 and Y != 0 where you would want to show individual stats for both because none of them would be a de facto background.

Regardless, this is pretty minor so I would go ahead and mark it as resolved and approve

Reply:
Uhh, you have seen the "-" on the left, right? That means it is not part of the code...